tests/utils/yaml.go

```diff
@@ -0,0 +1,91 @@
+package utils
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"path/filepath"
+
+	appsv1 "k8s.io/api/apps/v1"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	apiv1 "github.com/cloudnative-pg/cloudnative-pg/api/v1"
+)
+
+// CreateObjectsFromYAML creates kubernetes objects defined in a YAML file, in the namespace
+// NOTE: this function is meant to be used in Tests. It only recognizes a subset of
+//  kubernetes and CNPG object types, throwing errors for any other object types
+func CreateObjectsFromYAML(yamlPath, namespace string) ([]client.Object, error) {
+	data, err := ioutil.ReadFile(filepath.Clean(yamlPath))
+	if err != nil {
+		return nil, err
+	}
+
+	sections := bytes.Split(data, []byte("---"))
+	retVal := make([]client.Object, 0, len(sections))
+	decoder, err := getDecoder()
+	if err != nil {
+		return nil, err
+	}
+
+	for _, section := range sections {
+		if string(bytes.TrimSpace(section)) == "\n" || len(bytes.TrimSpace(section)) == 0 {
+			continue
+		}
+
+		obj, _, err := decoder.Decode(section, nil, nil)
+		if err != nil {
+			log.Printf("ERROR CreateObjectsFromYAML decoding: %s", err)
+			return nil, err
+		}
+		switch o := obj.(type) {
```

Comment:
Perhaps we could avoid this type switch here by using a `MetadataAccessor` (https://pkg.go.dev/k8s.io/apimachinery/pkg/api/meta#MetadataAccessor) or doing something similar to what we did here: https://github.com/cloudnative-pg/cloudnative-pg/blob/4eb7d0a79e291f97e4d1c9bc7504b4848e0ef50f/tests/utils/environment.go#L257 ?